### PR TITLE
Add applicationMenu to ConsoleLink CRD

### PR DESCRIPTION
**Pull request by benjaminapetersen**
Following these changes: https://github.com/openshift/api/pull/360

@spadgett @jhadvig @rhamilto

### CODE REVIEW

File: manifests/00-crd-extension-console-link.yaml [hunk 1]
```diff
@@ -54,3 +54,17 @@ spec:
...
+              required:
+                - section
+                - imageURL
+              properties:
+                section:
+                  type: string
+                  description: The section of the application menu in which the link should appear
+                imageURL:
+                  type: string
+                  description: The URL for the icon used in front of the link in the application menu.  The URL must be an HTTPS URL or a Data URI. The image should be square and will be shown at 24x24 pixels.
```
Comment (benjaminapetersen) [inline manifests/00-crd-extension-console-link.yaml]: @rhamilto does this default to anything?

Comment (benjaminapetersen) [reply manifests/00-crd-extension-console-link.yaml]: If it does, we don't need to list it as required.

Comment (spadgett) [reply manifests/00-crd-extension-console-link.yaml]: @jeff-phillips-18 

I don't think we should make it required. I'd guess it just doesn't show an icon.

File: manifests/00-crd-extension-console-link.yaml [hunk 2]
```diff
@@ -54,3 +54,17 @@ spec:
...
+
```
Comment (spadgett) [inline manifests/00-crd-extension-console-link.yaml]: nit: extra trailing newline

File: manifests/00-crd-extension-console-link.yaml [hunk 3]
```diff
@@ -54,3 +54,17 @@ spec:
               type: string
               description: Determines which dropdown menu the link is added to (HelpMenu, UserMenu)
               pattern: ^(HelpMenu|UserMenu)$
```
Comment (spadgett) [inline manifests/00-crd-extension-console-link.yaml]:
```suggestion
              pattern: ^(ApplicationMenu|HelpMenu|UserMenu)$
```

File: manifests/00-crd-extension-console-link.yaml [hunk 4]
```diff
@@ -54,3 +54,17 @@ spec:
...
+            applicationMenu:
+              type: object
+              description: Object that holds application menu defaults if the location is set to ApplicationMenu
```
Comment (spadgett) [inline manifests/00-crd-extension-console-link.yaml]: Are these really "defaults"? Maybe

```suggestion
              description: Object that holds application menu properties if the location is set to ApplicationMenu
```

File: manifests/00-crd-extension-console-link.yaml (hunk 1 above)
Comment (jhadvig) [reply manifests/00-crd-extension-console-link.yaml]: agree, lets dont make the `imageURL` required.

Comment (benjaminapetersen) [reply manifests/00-crd-extension-console-link.yaml]: Sounds good.

File: manifests/00-crd-extension-console-link.yaml (hunk 3 above)
Comment (benjaminapetersen) [reply manifests/00-crd-extension-console-link.yaml]: Excellent, can't miss this one.

File: manifests/00-crd-extension-console-link.yaml (hunk 4 above)
Comment (benjaminapetersen) [reply manifests/00-crd-extension-console-link.yaml]: Agree, much better.

File: manifests/00-crd-extension-console-link.yaml [hunk 5]
```diff
@@ -53,4 +53,16 @@ spec:
...
+            applicationMenu:
+              type: object
+              description: Object that holds application menu properties if the location is set to ApplicationMenu
+              required:
+                - section
+              properties:
+                section:
+                  type: string
+                  description: The section of the application menu in which the link should appear
+                imageURL:
```
Comment (spadgett) [inline manifests/00-crd-extension-console-link.yaml]: @benjaminapetersen @jhadvig do we need to add `+optional` to the property in openshift/api? I'm not aware of all the rules....

File: manifests/00-crd-extension-console-link.yaml [hunk 6]
```diff
@@ -53,4 +53,16 @@ spec:
             location:
               type: string
               description: Determines which dropdown menu the link is added to (HelpMenu, UserMenu)
```
Comment (spadgett) [inline manifests/00-crd-extension-console-link.yaml]: This comment needs to be updated as well.

File: manifests/00-crd-extension-console-link.yaml (hunk 5 above)
Comment (benjaminapetersen) [reply manifests/00-crd-extension-console-link.yaml]: Yes, we should.  I'll do a quick PR for that.

File: manifests/00-crd-extension-console-link.yaml (hunk 6 above)
Comment (benjaminapetersen) [reply manifests/00-crd-extension-console-link.yaml]: +1

**Comment by spadgett** [comment]
@jeff-phillips-18 when this merges, we should be able to make the corresponding console change to add in the links

**Pull request merged**